lib/quantizeAttributes.js

```diff
@@ -0,0 +1,348 @@
+            }
+
+            //TODO: Repack the buffers once, don't do it for each accessor
+            //Adjust buffer to be float aligned
+            var shift = (encoded.length * 2) % 4;
+
+            var bufferId = accessorData.bufferId;
+            var buffer = accessorData.buffer;
+            var source = buffer.extras._pipeline.source;
+            var offset = accessorData.byteOffset;
+            var length = accessorData.byteLength;
+            var byteSizeDifference = length - (encoded.length * 2 + shift);
+
+            var output = new Uint8Array(buffer.byteLength - byteSizeDifference);
+            output.set(source.slice(0, offset));
+            output.set(new Uint8Array(encoded.buffer), offset);
+            output.set(source.slice(offset + length), offset + encoded.length * 2 + shift);
+
+            accessor.byteStride = accessor.byteStride / 2;
+            accessor.componentType = 5123;
+
+            // Overwrite the buffer
+            gltf.buffers[accessorData.bufferId] = {
+                type: "arraybuffer",
+                byteLength: output.length,
+                extras: {
+                    _pipeline: {
+                        source: output,
+                        deleteExtras: true,
+                        extension: buffer.extras._pipeline.extension
+                    }
+                }
+            };
+
+            // Correct the bufferview
+            var bufferView = accessorData.bufferView;
+            bufferView.byteLength -= byteSizeDifference;
+
+            //correct offsets and for any buffer views that use the same buffer
+            for (var bufferViewId in gltf.bufferViews) {
+                if (gltf.bufferViews.hasOwnProperty(bufferViewId)) {
+                    bufferView = gltf.bufferViews[bufferViewId];
+                    if (bufferView.buffer === bufferId) {
+                        if (bufferView.byteOffset > offset) {
+                            bufferView.byteOffset -= byteSizeDifference;
+                        }
+                    }
+                }
+            }
+
+            //correct offsets for any accessors that use the same bufferview
+            for (var otherAccessorId in accessors) {
+                if (accessors.hasOwnProperty(otherAccessorId)) {
+                    var otherAccessor = accessors[otherAccessorId];
+                    if (otherAccessor.bufferView === accessor.bufferView) {
+                        if (otherAccessor.byteOffset > accessor.byteOffset) {
+                            otherAccessor.byteOffset -= byteSizeDifference;
+                        }
+                    }
+                }
+            }
+        }
+    }
+    // Add WEB3D_quantized_attributes extension to extensionsUsed
+    if (!defined(extensionsUsed)) {
+        extensionsUsed = [];
+        gltf.extensionsUsed = extensionsUsed;
+    }
+    extensionsUsed.push('WEB3D_quantized_attributes');
```

Comment:
Perhaps only add this if it doesn't already exist (make a helper function like `addExtensionsUsed(gltf, 'WEB3D_quantized_attributes')` that does all the required checks.